src/Lecture2.hs

```diff
@@ -97,7 +109,7 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+dropSpaces = unwords . words
```

Comment:
I see that this implementation hangs on the list that contains an infinite number of spaces after the word:

```haskell
ghci> unwords (words ("foo" ++ repeat ' '))
```

Also, this function lacks a top-level type signature 🔝

Reply:
I'm unused to thinking about the infinite case 😅

Reply:
Do you have any pointers for this?

Reply:
You can use functions `dropWhile` and `takeWhile` to drop all spaces first and then take all non-spaces. Both these functions are lazy and `takeWhile` will stop on first space character making this function lazy 🙂